components/batch-system/src/batch.rs

```diff
@@ -377,30 +392,51 @@ where
         &self.router
     }
 
+    fn start_poller<B>(&mut self, name: String, priority: Priority, builder: &mut B)
+    where
+        B: HandlerBuilder<N, C>,
+        B::Handler: Send + 'static,
+    {
+        let handler = builder.build(priority);
+        let receiver = match priority {
+            Priority::Normal => self.receiver.clone(),
+            Priority::Low => self.low_receiver.clone(),
+        };
+        let mut poller = Poller {
+            router: self.router.clone(),
+            fsm_receiver: receiver,
+            handler,
+            max_batch_size: self.max_batch_size,
+            reschedule_duration: self.reschedule_duration,
+        };
+        let t = thread::Builder::new()
+            .name(name)
+            .spawn(move || {
+                set_io_type(IOType::ForegroundWrite);
+                poller.poll();
+            })
+            .unwrap();
+        self.workers.push(t);
+    }
+
     /// Start the batch system.
     pub fn spawn<B>(&mut self, name_prefix: String, mut builder: B)
     where
         B: HandlerBuilder<N, C>,
         B::Handler: Send + 'static,
     {
         for i in 0..self.pool_size {
-            let handler = builder.build();
-            let mut poller = Poller {
-                router: self.router.clone(),
-                fsm_receiver: self.receiver.clone(),
-                handler,
-                max_batch_size: self.max_batch_size,
-                reschedule_duration: self.reschedule_duration,
-            };
-            let t = thread::Builder::new()
-                .name(thd_name!(format!("{}-{}", name_prefix, i)))
-                .spawn(move || {
-                    set_io_type(IOType::ForegroundWrite);
-                    poller.poll()
-                })
-                .unwrap();
-            self.workers.push(t);
+            self.start_poller(
+                thd_name!(format!("{}-{}", name_prefix, i)),
+                Priority::Normal,
+                &mut builder,
+            );
         }
+        self.start_poller(
+            thd_name!(format!("{}-low", name_prefix)),
+            Priority::Low,
+            &mut builder,
+        );
```

Comment:
The store thread pool doesn't need a low-priority thread. I suggest adding a config to fix it.

Reply:
Emmmm, but I want to move the regions which have been a long time no processing data to low-priority thread so that their heartbeat can be processed independently.